library/init/meta/widget/basic.lean

```diff
@@ -279,19 +279,30 @@ meta def with_cn : string → html α → html α
 meta def with_key {β} [has_to_string β] : β → html α → html α
 | s h := with_attr (key s) h
 
+/-- An effect is an action at the root of the widget component hierarchy
+and can give instructions to the editor to perform some task. -/
+meta inductive effect : Type
+| insert_text (text : string)
+| reveal_position (file_name : string) (p : pos)
```

Comment:
Is empty string = current file?  We cannot get the file name of the current file in meta code.